src/block_disagg/block_disagg_write.c

```diff
@@ -116,17 +128,50 @@ __wt_block_disagg_write_internal(WT_SESSION_IMPL *session, WT_BLOCK_DISAGG *bloc
      */
     blk->flags = 0;
     if (data_checksum)
-        F_SET(blk, WT_BLOCK_DATA_CKSUM);
+        F_SET(blk, WT_BLOCK_DISAGG_DATA_CKSUM);
+
+    /*
+     * XXX temporary measure until we the block header at the beginning of the data. Set the block
+     * manager encryption/compression flags - this is eventually to be done by our caller.
+     */
+    if (!is_delta) {
+        page_header = (WT_PAGE_HEADER *)buf->mem;
+        if (F_ISSET(page_header, WT_PAGE_COMPRESSED))
+            F_SET(blk, WT_BLOCK_DISAGG_COMPRESSED);
+        if (F_ISSET(page_header, WT_PAGE_ENCRYPTED))
+            F_SET(blk, WT_BLOCK_DISAGG_ENCRYPTED);
+    }
+
+    if (block_meta->delta_count == 0) {
+        blk->magic = WT_BLOCK_DISAGG_MAGIC_BASE;
+        blk->header_size = WT_BLOCK_DISAGG_BASE_HEADER_BYTE_SIZE;
+    } else {
+        blk->magic = WT_BLOCK_DISAGG_MAGIC_DELTA;
+        blk->header_size = WT_BLOCK_DISAGG_DELTA_HEADER_BYTE_SIZE;
+        F_SET(&put_args, WT_PAGE_LOG_DELTA);
+    }
+    /* blk->version and blk->compatiable_version are implicitly 0. */
```

Comment:
Ah, this is why I thought the compatible version was unused - typo here so my ctrl+f didn't find it. I'm still interested in why we want it though.